lib/optimizer.py

```diff
@@ -151,6 +153,39 @@ def BestUntriedEncoding(self, bitrate, videofile, hashnames_to_ignore=None):
         return encoding
     return None
 
+  def BestOverallEncoder(self):
+    """Returns the configuration that is best over all files.
+
+    This looks only at configurations that have been run for every
+    file and rate in the fileset."""
+    files_and_rates = self.file_set.AllFilesAndRates()
+    candidate_encoders = set([x.encoder.Hashname()
+                          for x in self.context.cache.AllScoredEncodings(
+                              files_and_rates[0][0],
+                              encoder.Videofile(files_and_rates[0][1]))])
+    for rate, filename in files_and_rates[1:]:
+      these_encoders = set([x.encoder.Hashname()
+                            for x in self.context.cache.AllScoredEncodings(
+                              rate, encoder.Videofile(filename))])
+      candidate_encoders.intersection_update(these_encoders)
+    if len(candidate_encoders) == 0:
+      return None
+    if len(candidate_encoders) == 1:
```

Comment:
Here though you're checking that the length is precisely one, so you can't to if candidate_encoders.